pypesto/storage/save_to_hdf5.py

```diff
@@ -58,23 +48,18 @@ def write(self, problem, overwrite: bool = False):
                                     "information about optimization result."
                                     "If you wish to overwrite the file set"
                                     "overwrite=True.")
+            attrs_to_save = [a for a in dir(problem) if not a.startswith('__')
+                             and not callable(getattr(problem, a))]
 
             problem_grp = f.create_group("problem")
             # problem_grp.attrs['config'] = objective.get_config()
-            problem_grp.attrs[self.DIM] = problem.dim
-            problem_grp.attrs[self.DIM_FULL] = problem.dim_full
-
-            write_float_array(problem_grp, self.LB, problem.lb)
-            write_float_array(problem_grp, self.UB, problem.ub)
-            write_float_array(problem_grp, self.LB_FULL, problem.lb_full)
-            write_float_array(problem_grp, self.UB_FULL, problem.ub_full)
-            write_float_array(problem_grp, self.X_FIXED_VALS,
-                              problem.x_fixed_vals)
-            write_int_array(problem_grp, self.X_FIXED_INDICES,
-                            problem.x_fixed_indices)
-            write_int_array(problem_grp, self.X_FREE_INDICES,
-                            problem.x_free_indices)
-            write_string_array(problem_grp, self.X_NAMES, problem.x_names)
+
+            for problem_attr in attrs_to_save:
+                value = getattr(problem, problem_attr)
+                if isinstance(value, (list, np.ndarray)):
+                    write_array(problem_grp, problem_attr, value)
+                elif isinstance(value, int):
```

Comment:
numbers.Integral, as above?

Reply:
The automatic type deduction may lead to funny situations, where the user passes an `int` for a value that is usually a float. May be unexpected to have varying HDF5 datatypes for the same attributes depending on whether one passes 1 or 1.0. As long as only accessed from python, i guess it won't matter too much... Fine to leave...

Reply:
float->int shouldn't be a problem. as long as int->int is fulfilled, everything should work. specifying the type would only be possible with providing an explicit list of attributes I guess.